driver/normalizer/normalizer.go

```diff
@@ -1055,6 +1236,64 @@ func (op opMergeGroups) Check(st *State, n nodes.Node) (bool, error) {
 	return op.sub.Check(st, fgroup)
 }
 
+// checkFuncGroup tests if the current node is uast:FuncGroup and if any of its sub-arrays
+// contain a ust:Group, it will be removed and the children will be flattened into a sub-array.
+func (op opMergeGroups) checkFuncGroup(st *State, fgroup nodes.Object) (bool, error) {
+	// primary nodes array in the function group
+	arr, ok := fgroup["Nodes"].(nodes.Array)
+	if !ok {
+		return false, errors.New("expected an array in FuncGroup.Nodes")
+	}
+	modified := false
+	for i := 0; i < len(arr); i++ {
+		v := arr[i]
+		if v == nil {
+			// remove nil elements
```

Comment:
Is removing `nil` a common feature of all checks, or just this one? If it's common, I suggest pulling this filter out into a separate function. These lists are not long, and it would be no big deal to take a second pass over them. If it's idiosyncratic to this check, could you please mention that aspect in the comment too?